### PR TITLE
chore: Moved code machers from Codux

### PR DESCRIPTION
**Pull request by daomry**
- Moved code machers from Codux repo
- code matchers use prettier 3 (async)
- retry() fix for code matchers
- removed `formatted` matcher prop support

### CODE REVIEW

File: packages/testing/src/code-matchres.types.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+            /**
+             * @see matchCode
+             */
+            matchesCode(code: string): PromiseLikeAssertion;
+
+            /**
+             * Asserts that the expression that will be tested includes the expected expression.
+             * Both expected and actual expressions are formatted using prettier.
+             *
+             * If you don't want the expected expression to be formatted, use `.formatted.to.include()`
```
Comment (vladyslav-baliuk) [inline packages/testing/src/code-matchres.types.ts:22]: Is this part of the comment still relevant? As I see we aren't supporting `.formatted` assertion property anymore

File: packages/testing/src/test/code-matcher.unit.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    });
+    it('assert equivalent code is not contained', async () => {
+        await expect(MULTILINE).not.to.includeCode(`const NOT='included';`);
+    });
+    it('throws when code is not a equivalent', async () => {
+        await expect(expect(MULTILINE).to.includeCode(`const some="random code";`)).to.be.rejectedWith(
+            'Expected code to include',
+        );
+    });
+});
```
Comment (vladyslav-baliuk) [inline packages/testing/src/test/code-matcher.unit.ts:60]: missing empty line at the end of the file

File: packages/testing/src/code-matchers.ts [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        parser: 'typescript',
+        plugins: [esTreePlugin, parserTypeScript],
+        endOfLine: 'lf',
+        singleQuote: true,
+    });
+
+const regex = /(\/\*)([^\/\*]*)(\*\/)/g;
+
+const mapper = (str: string, idx: number) => {
+    if (idx % 4 === 2) {
```
Comment (vladyslav-baliuk) [inline packages/testing/src/code-matchers.ts:20]: Optional: check the negative condition first to reduce indentation, like this:

```
if (idx % 4 !== 2) return str;

return str.split...
```

**Review by vladyslav-baliuk — APPROVED** [review]

File: packages/testing/src/code-matchers.ts (hunk 3 above)
Comment (daomry) [reply packages/testing/src/code-matchers.ts:20]: This code is lifted from Codux, I prefer not to change it where possible (just to make the change more frictionless)

**Pull request merged**